api/routes/notesRouter.js

```diff
@@ -0,0 +1,169 @@
+const notes = require("../../notes/notesModel");
+const tags = require("../../notes/tagsModel");
+
+const express = require("express");
+const router = express.Router();
+
+const { authenticate } = require("../../auth/authenticate");
+
+const requestOptions = {
+  headers: { accept: "application/json" }
+};
+
+router.get("/", (req, res) => {
+  notes
+    .fetch()
+    .then(notes => {
+      notes[0]
+        ? res.json(notes)
+        : res
+            .status(400)
+            .json({ error: "there are currently no notes in our directory" });
+    })
+    .catch(err => {
+      res.status(500).json({ error: "could not retrieve notes" });
+    });
+});
+
+router.get("/:id", (req, res) => {
+  const { id } = req.params;
+  notes
+    .fetch(id)
+    .then(note => {
+      if (note[0]) {
+        res.json(note);
+      } else {
+        res.status(404).json({ error: "note does not exist" });
+      }
+    })
+    .catch(err => {
+      res.status(500).json({ error: "note could not be retrieved." });
+    });
+});
+
+router.get("/:id/tags", (req, res) => {
+  const { id } = req.params;
+
+  notes
+    .fetch(id)
+    .then(note => {
+      if (note[0]) {
+        tags.fetchTagsByNote(note[0].id).then(tags => {
+          if (tags[0]) {
+            res.json(tags);
+          } else {
+            res.status(404).json({ error: "note has no tags" });
+          }
+        });
+      } else {
+        res.status(404).json({ error: "note does not exist" });
+      }
+    })
+    .catch(err => {
+      res.status(500).json({ error: "note tags could not be retrieved." });
+    });
+});
+
```

Comment:
Your code is in great shape Angela 👍  because you're being so thorough with your error messages think about extracting them to another file and placing them in an object, then you can inject them throughout your app, and easily add or change them.
```
export const errMessages = { 
    noNoteFound: 'note does not exist.',
    noNoteTags: 'note tags could not be retrieved.',
    etc....
}
```